pkg/storage/ingest/writer.go

```diff
@@ -192,9 +207,10 @@ func (w *Writer) produceSync(ctx context.Context, client *kgo.Client, records []
 	// Wait for a response or until the context has done.
 	select {
 	case <-ctx.Done():
-		return context.Cause(ctx)
+		return kgo.ProduceResults{{Err: context.Cause(ctx)}}
```

Comment:
`<-ctx.Done` returns also on deadline exceeded error, in that case `context.Cause(ctx)` will be nil. Shall we handle that?

Reply:
I don't fully understand the concern. If there's no cause, `context.Cause()` returns `ctx.Err()`. It just look safer to call `context.Cause()` to me, given `ctx` is passed in input to `WriteSync()` and the caller may cancel it with a cause.

Reply:
> I don't fully understand the concern. If there's no cause, `context.Cause()` returns `ctx.Err()`.

I got confused by documentation to `context.Cause`, which says that `Cause returns nil if c has not been canceled yet.`. I read it as `context.Cause` will return `nil` also for `context.DeadlineExceeded`, which is not what we want.

But it turns out I misunderstood the documentation.